lib/utils/string_extension.dart

```diff
@@ -373,4 +373,24 @@ extension StringCasingExtension on String {
     final match = regex.firstMatch(this);
     return match?.group(1);
   }
+
+  bool isRoomAlias() {
+    final regExp = RegExp(r'^[#][^:]+:.+$');
+    final result = regExp.hasMatch(trim());
+    return result;
+  }
+
+  String? getServerNameFromRoomIdOrAlias() {
+    final parts = split(':');
+    if (parts.length > 1) {
+      return parts[1];
+    }
+    return null;
+  }
+
+  bool isRoomId() {
+    final regExp = RegExp(r'^[!][^:]+:.+$');
```

Comment:
IMO you should you isValidMatrixId instead from matrix dart sdk

https://github.com/linagora/matrix-dart-sdk/blob/272cb73ab18cff7ae534ef4eafa61c6a3494cb00/lib/src/utils/matrix_id_string_extension.dart#L33-L50

Reply:
also it won't work for `!localpart:domain:port`